libvcx/src/libindy/proofs/proof_request_internal.rs

```diff
@@ -0,0 +1,48 @@
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct Filter {
+    pub schema_id: Option<String>,
+    pub schema_issuer_did: Option<String>,
+    pub schema_name: Option<String>,
+    pub schema_version: Option<String>,
+    pub issuer_did: Option<String>,
+    pub cred_def_id: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(untagged)]
+pub enum Restrictions {
+    V1(Vec<Filter>),
+    V2(::serde_json::Value),
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct PredicateInfo {
+    pub name: String,
+    //Todo: Update p_type to use Enum
+    pub p_type: String,
+    pub p_value: i32,
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
I wonder why is `skip_serializing_if` necessary?

Reply:
Well, I haven't done modifications here, just moved the datastructure, however I suppose it makes sense to skip serializing the field if there's no value set.

Reply:
But why is it necessary? Extra fields are ignored during deserialization...

Reply:
```
use serde_json;

#[derive(Serialize)]
struct Data {
    #[serde(skip_serializing_if = "Option::is_none")]
    a: Option<String>,
    c: Option<String>,
}

pub fn run()
{
    let data = Data { a:Some("sad".into()), c:None};
    let serialized = serde_json::to_string(&data).unwrap();
    println!("serialized={}", serialized);
}
```

```
serialized={"a":"sad","c":null}
```

Reply:
I don't see what you mean by this (I said "deserialization", not "serialization" :))...